lib/lhm/retry_helper.rb

```diff
@@ -3,36 +3,43 @@
 
 module Lhm
   # RetryHelper standardizes the interface for retry behavior in components like
-  # Entangler and AtomicSwitcher.
+  # Entangler, AtomicSwitcher, ChunkerInsert.
   #
-  # To retry some behavior, use `execute_with_retries(statement)`
-  # which assumes `@connection` is available.
+  # To retry some behavior, use `connection_with_retries(statement: sql, invoke_with: :execute)`.
+  # Retry helper assumes:
+  # * `@connection` is available
+  # * `configure_retry` is called before `connection_with_retries` is used
   #
-  # `execute_with_retries` expects the caller to invoke `configure_retry` first, providing:
-  # * `tries` as an integer
-  # * `base_interval` as an integer
+  # If an error includes the message "Lock wait timeout exceeded",
+  # the RetryHelper will retry the SQL command again after about 500ms
+  # for up to one hour.
   #
-  # For a full list of configuration options see https://github.com/kamui/retriable
+  # This behavior can be modified by calling `configure_retry` with options described in
+  # https://github.com/kamui/retriable
   module RetryHelper
-    def execute_with_retries(statement)
+    def connection_with_retries(statement:, invoke_with:)
       Retriable.retriable(retry_config) do
-        @connection.execute(SqlHelper.tagged(statement))
+        @connection.public_send(invoke_with, SqlHelper.tagged(statement))
       end
     end
 
     def configure_retry(options)
-      @retry_config = DEFAULT_RETRY_CONFIG.merge(options)
+      @retry_config = DEFAULT_RETRY_CONFIG.dup
+      @retry_config.merge!(options) if options
     end
 
     attr_reader :retry_config
 
     private
 
+    # For a full list of configuration options see https://github.com/kamui/retriable
     DEFAULT_RETRY_CONFIG = {
       on: {
         StandardError => [/Lock wait timeout exceeded/]
```

Comment:
We also need to retry on deadlocks, something that may affect the chunker on tables with UNIQUE keys (besides the primary) with high rates of writes, which are precisely our most painful ones.